### PR TITLE
Entry for Swift Comparable protocol

### PR DESCRIPTION
**Pull request by aesiyok**
### Description
Closes#2882

Adds a new entry for the Swift Comparable protocol in the file docs/content/swift/concepts/protocols/terms/comparable/comparable.md

The entry includes a description of the protocol and its requirements, the syntax, and an example of it in use.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/swift/concepts/protocols/terms/comparable/comparable.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
+Title: 'Comparable Protocol' # Required; the file name should be the same as the title, but lowercase, with dashes instead of spaces, and all punctuation removed
+Description: 'Protocol in Swift used to define a sort order for instances of a type' # Required; ideally under 150 characters and starts with a present-tense verb (used in search engine results and content previews)
```
Comment (SSwiniarski) [inline content/swift/concepts/protocols/terms/comparable/comparable.md:3]:
```suggestion
Description: 'Protocol in Swift used to define a sort order for instances of a type.' # Required; ideally under 150 characters and starts with a present-tense verb (used in search engine results and content previews)
```

File: content/swift/concepts/protocols/terms/comparable/comparable.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+Subjects: # Please only use Subjects in the subjects.md file (https://github.com/Codecademy/docs/blob/main/documentation/subjects.md). If that list feels insufficient, feel free to create a new Subject and add it to subjects.md in your PR!
+  - 'Mobile Development'
+  - 'Computer Science'
+Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!
+  - 'Protocols'
```
Comment (SSwiniarski) [inline content/swift/concepts/protocols/terms/comparable/comparable.md:8]: This tag is not in the list here: https://github.com/Codecademy/docs/blob/main/documentation/tags.md

Please use tags from that list.

File: content/swift/concepts/protocols/terms/comparable/comparable.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
```
Comment (SSwiniarski) [inline content/swift/concepts/protocols/terms/comparable/comparable.md:9]:
```suggestion
```
Delete extra line

File: content/swift/concepts/protocols/terms/comparable/comparable.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
+  - 'learn-swift'
+  - 'paths/build-ios-apps-with-swiftui'
+---
+
+The **Comparable** protocol in Swift is used to define a sort order for instances of a type.
+> To add Comparable conformance to your custom types, define the < and == operators as static methods of your types.
```
Comment (SSwiniarski) [inline content/swift/concepts/protocols/terms/comparable/comparable.md:16]:
```suggestion
The **`Comparable`** protocol in Swift is used to define a sort order for instances of a type. To add `Comparable` conformance to custom types, define the < and == operators as static methods of those types.
```

File: content/swift/concepts/protocols/terms/comparable/comparable.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+## Syntax
+
+
+
```
Comment (SSwiniarski) [inline content/swift/concepts/protocols/terms/comparable/comparable.md:21]:
```suggestion

```
Only one blank line.

File: content/swift/concepts/protocols/terms/comparable/comparable.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+```
+struct StructName: Comparable {
+    let var: Int
+
+    static func < (lhs: StructName, rhs: StructName) -> Bool {
+    return lhs.var < rhs.var
+    }
+
+    static func == (lhs: Building, rhs: Building) -> Bool {
```
Comment (SSwiniarski) [inline content/swift/concepts/protocols/terms/comparable/comparable.md:30]:
```suggestion
    static func == (lhs: StructName, rhs: StructName) -> Bool {
```

File: content/swift/concepts/protocols/terms/comparable/comparable.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    return lhs.var == rhs.var
+    }
+}
+```
+
```
Comment (SSwiniarski) [inline content/swift/concepts/protocols/terms/comparable/comparable.md:35]: We generally have a sentence or two describing what the syntax is showing.

File: content/swift/concepts/protocols/terms/comparable/comparable.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+## Example
+
```
Comment (SSwiniarski) [inline content/swift/concepts/protocols/terms/comparable/comparable.md:37]: We generally have a sentence introducing the example code and summarizing what it is doing.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@aesiyok I did a first review and left a few comments.

File: content/swift/concepts/protocols/terms/comparable/comparable.md (hunk 2 above)
Comment (aesiyok) [reply content/swift/concepts/protocols/terms/comparable/comparable.md]: Changed tag to 'iOS'

File: content/swift/concepts/protocols/terms/comparable/comparable.md (hunk 7 above)
Comment (aesiyok) [reply content/swift/concepts/protocols/terms/comparable/comparable.md:31]: Added sentences describing the syntax.

File: content/swift/concepts/protocols/terms/comparable/comparable.md (hunk 8 above)
Comment (aesiyok) [reply content/swift/concepts/protocols/terms/comparable/comparable.md:33]: Added sentences describing the example. Thank you for reviewing this entry. I have made all of your suggested changes.

**Comment by aesiyok** [comment]
> @aesiyok I did a first review and left a few comments.

Thanks for reviewing my entry. I have made all of the suggested changes.

**Review by SSwiniarski — APPROVED** [review]
@aesiyok I did one minor change and this looks good for a second review.

File: content/swift/concepts/protocols/terms/comparable/comparable.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+  - 'Mobile Development'
+  - 'Computer Science'
+Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!
+  - 'iOS'
+CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
+  - 'learn-swift'
+  - 'paths/build-ios-apps-with-swiftui'
+---
+
+The **`Comparable`** protocol in Swift is used to define a sort order for instances of a type. To add `Comparable` conformance to custom types, define the < and == operators as static methods of those types.
```
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/comparable/comparable.md:14]:
```suggestion
The **`Comparable`** protocol in Swift is used to define a sort order for instances of a type. To add `Comparable` conformance to custom types, define the `<` and `==` operators as static methods of those types.
```

File: content/swift/concepts/protocols/terms/comparable/comparable.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+## Syntax
```
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/comparable/comparable.md:16]: add an empty line after the header

File: content/swift/concepts/protocols/terms/comparable/comparable.md [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+
+    static func < (lhs: StructName, rhs: StructName) -> Bool {
+    return lhs.var < rhs.var
+    }
+
+    static func == (lhs: StructName, rhs: StructName) -> Bool {
+    return lhs.var == rhs.var
+    }
+}
+```
```
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/comparable/comparable.md:29]: add an empty line after the code block

File: content/swift/concepts/protocols/terms/comparable/comparable.md [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+To conform a custom type to the **Comparable** protocol conformance to **Comparable** is first declared, then static methods for < and == are defined.
```
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/comparable/comparable.md:30]:
```suggestion
To conform a custom type to the `Comparable` protocol conformance to `Comparable` is first declared, then static methods for `<` and `==` are defined.
```

File: content/swift/concepts/protocols/terms/comparable/comparable.md [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+    static func == (lhs: Building, rhs: Building) -> Bool {
+    return lhs.height == rhs.height
+    }
+}
+
+let eiffelTower = Building(height: 300)
+let burjKhalifa = Building(height: 828)
+let sydneyTower = Building(height: 300)
+```
+A custom type `Building` was created with a `height` constant, and static methods for < and == were defined. There are three instances of this type: `eiffelTower`, `BurjKhalifa` and `sydneyTower`. The example conforms to the `Comparable` protocol, allowing each of the instances to be compared:
```
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/comparable/comparable.md:51]:
```suggestion
A custom type `Building` was created with a `height` constant, and static methods for `<` and `==` were defined. There are three instances of this type: `eiffelTower`, `burjKhalifa` and `sydneyTower`. The example conforms to the `Comparable` protocol, allowing each of the instances to be compared:
```

Comment (caupolicandiaz) [reply content/swift/concepts/protocols/terms/comparable/comparable.md:51]: add an empty line between the code and this section

File: content/swift/concepts/protocols/terms/comparable/comparable.md [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+    return lhs.var == rhs.var
+    }
+}
+```
+To conform a custom type to the **Comparable** protocol conformance to **Comparable** is first declared, then static methods for < and == are defined.
+
+## Example
+
+```
+struct Building: Comparable {
```
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/comparable/comparable.md:35]: the code block should use 2 spaces, not tabs, for indentations

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@aesiyok I've completed my initial review. Please update the entry, and tag me when you've finished making changes. Thanks

**Comment by aesiyok** [comment]
@caupolicandiaz Thanks for reviewing my entry, I have made the necessary changes.

**Review by caupolicandiaz — APPROVED** [review]
@aesiyok @SSwiniarski Looks good to merge.

**Pull request merged**